### PR TITLE
[layer] Update preprocess layers for layerV2 

### PR DESCRIPTION
**Pull request by kparichay**
Update preprocess layers for layerv2 design.
Enable corresponding unittests.

Signed-off-by: Parichay Kapoor <kparichay@gmail.com>

### CODE REVIEW

File: nntrainer/layers/preprocess_flip_layer.cpp [hunk 1]
```diff
@@ -26,55 +27,80 @@ const std::string PreprocessFlipLayer::flip_vertical = "vertical";
...
   }
 }
 
-void PreprocessFlipLayer::forwarding(bool training) {
+void PreprocessFlipLayer::forwarding(RunLayerContext &context, bool training) {
   if (!training) {
-    for (unsigned int idx = 0; idx < input_dim.size(); idx++) {
-      net_hidden[idx]->getVariableRef() = net_input[idx]->getVariableRef();
+    for (unsigned int idx = 0; idx < context.getNumInputs(); idx++) {
+      context.getOutput(idx) = context.getInput(idx);
```
Comment (zhoonit) [inline nntrainer/layers/preprocess_flip_layer.cpp:87]: not related to the PR but is this allowed in the point of graph?

File: nntrainer/layers/preprocess_translate_layer.cpp [hunk 2]
```diff
@@ -45,75 +47,100 @@ int PreprocessTranslateLayer::initialize(Manager &manager) {
...
 }
 
-void PreprocessTranslateLayer::forwarding(bool training) {
+void PreprocessTranslateLayer::forwarding(RunLayerContext &context,
+                                          bool training) {
   if (!training) {
-    for (unsigned int idx = 0; idx < input_dim.size(); idx++) {
-      net_hidden[idx]->getVariableRef() = net_input[idx]->getVariableRef();
+    for (unsigned int idx = 0; idx < context.getNumInputs(); idx++) {
+      context.getOutput(idx) = context.getInput(idx);
```
Comment (zhoonit) [inline nntrainer/layers/preprocess_translate_layer.cpp:107]: same for the record

File: nntrainer/layers/preprocess_flip_layer.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/preprocess_flip_layer.cpp:87]: Ideally this should do copy or set supportInPlace() to true.
I will add a todo to check this later.

File: nntrainer/layers/preprocess_translate_layer.cpp (hunk 2 above)
Comment (kparichay) [reply nntrainer/layers/preprocess_translate_layer.cpp:107]: same reply as https://github.com/nnstreamer/nntrainer/pull/1386#discussion_r669254383

File: nntrainer/layers/preprocess_translate_layer.cpp [hunk 3]
```diff
@@ -45,75 +47,100 @@ int PreprocessTranslateLayer::initialize(Manager &manager) {
     affine_transform_mat.at<float>(1, 1) = 1;
 
     input_mat =
-      cv::Mat::zeros(input_dim[0].height(), input_dim[0].width(), CV_32FC3);
+      cv::Mat::zeros(input_dim_0.height(), input_dim_0.width(), CV_32FC3);
     output_mat =
-      cv::Mat::zeros(input_dim[0].height(), input_dim[0].width(), CV_32FC3);
+      cv::Mat::zeros(input_dim_0.height(), input_dim_0.width(), CV_32FC3);
```
Comment (jijoongmoon) [inline nntrainer/layers/preprocess_translate_layer.cpp:52]: This is not related with this PR, but add todo here ( implementation without opencv )

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/layers/preprocess_translate_layer.cpp (hunk 3 above)
Comment (kparichay) [reply nntrainer/layers/preprocess_translate_layer.cpp:52]: I will add.

File: nntrainer/layers/preprocess_flip_layer.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/preprocess_flip_layer.cpp:87]: This just does not use the given memory and works differently for train/inference.
Maybe `supportInPlace` should take `ExecutionMode` as an argument.

File: nntrainer/layers/preprocess_translate_layer.cpp (hunk 3 above)
Comment (kparichay) [reply nntrainer/layers/preprocess_translate_layer.cpp:52]: done.

**Review by zhoonit — APPROVED** [review]
lgtm

File: nntrainer/layers/preprocess_flip_layer.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/layers/preprocess_flip_layer.cpp:88]: I'll keep in mind :)

**Pull request merged**